### PR TITLE
[21783] Add notes for Fast DDS `v3.1.0`

### PR DESCRIPTION
**Pull request by Mario-DL**
## Contributor Checklist

- [X] Commit messages follow the project guidelines. 
- **N/A** Code snippets related to the added documentation have been provided.
- [x] Documentation tests pass locally.
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] CI passes without warnings or errors.

### CODE REVIEW

File: docs/notes/previous_versions/v3.1.0.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#. Complete support for ``Dynamic Network Interfaces``
+#. Add QoS getters from raw ``XML``
+
+This release includes the following **improvements**:
+
+#. Be less strict with parameter lengths
+#. Modify help command of ``XML CLI tool``
+#. Register and propagate ``MonitorService`` status type object
+#. Add tests for ``TypeLookup service``
+#. Convert ``Fast DDS v2`` to ``v3`` Migration Changes into a Step-by-Step User Guide
```
Comment (JesusPoderoso) [inline docs/notes/previous_versions/v3.1.0.rst:17]: I would add a link in the migration guide section and add a reference to the section here
```suggestion
#. Convert ``Fast DDS v2`` to ``v3`` Migration Changes into a :ref:`Step-by-Step User Guide <migration_guide>`.
```

File: docs/notes/previous_versions/v3.1.0.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
+`Version 3.1.0 (latest) <https://fast-dds.docs.eprosima.com/en/v3.1.0/index.html>`_
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This minor release includes the following **features** in an ABI compatible manner:
+
+#. Allow runing CI on external contributions
+#. Allow for all durability configurations
+#. Complete support for ``Dynamic Network Interfaces``
+#. Add QoS getters from raw ``XML``
```
Comment (JesusPoderoso) [inline docs/notes/previous_versions/v3.1.0.rst:9]: Can we add any documentation reference here?

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
We forgot to update the Dependencies and compatibilities section in the Bump to 3.1.x with the new tabs.
I think we should include it in this PR, together with the new versions of CDR, Gen and Shapes

File: docs/notes/previous_versions/v3.1.0.rst [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#. Fix ``flow_controllers`` comparison in ``DomainParticipantQos`` equality operator
+#. Fix ``CreateTopicWithDifferentTypeName_negative`` flaky test
+#. Warn instead of fail when ``dds.communication.dynamic_interfaces`` cannot be built
+#. Fix wrong Data type referred in ``flow control`` example ``README``
+
+.. important::
+
+    When upgrading to version 3.1.0 it is **required** to regenerate generated source from IDL files
+    using at least `Fast DDS-Gen v4.0.0 <https://github.com/eProsima/Fast-DDS-Gen/releases/tag/v4.0.0>`_.
+    But it is advisable to regenerate them using the latest patch version of ``Fast DDS-Gen v4.0.x``.
```
[the remark this answers is not in the document]
Comment (Mario-DL) [reply docs/notes/previous_versions/v3.1.0.rst:47]: Okay, we mention that it is advisable to use the latest patch version, but I aggree that it is more complete to state directly the minimum to `2`

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Pull request merged**